scripts/release.sh

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+tmp=$(mktemp -d)
+trap 'rm -rf "$tmp"' exit
+
+nixeval() {
+  # Since there's no stable `nix eval --raw`, also see https://github.com/NixOS/nix/pull/9361
+  nix-instantiate --eval --json "$@" | jq -r .
+}
+
+# The system to pre-build the release for and distribute artifacts for
+system=x86_64-linux
+root=$(git rev-parse --show-toplevel)
+repository=${GITHUB_REPOSITORY:-NixOS/nixpkgs-check-by-name}
+
+# Get the version from the Cargo.toml file
+version=$(nixeval "$root" -A build.version)
+echo "Current version is $version"
+
+if existingRelease=$(gh api \
+  -H "Accept: application/vnd.github+json" \
+  -H "X-GitHub-Api-Version: 2022-11-28" \
+  /repos/"$repository"/releases/tags/"$version"); then
+  echo "Release $version already exists, no new release necessary"
+  exit 0
+else
+  echo "$existingRelease"
+  echo "Release $version doesn't exist yet, creating it"
+fi
+
+echo "Building release artifact for system $system"
+
+nix-build "$root" -A build -o "$tmp/build" >/dev/null
+readarray -t closure < <(nix-store -qR "$tmp/build")
+nix-store --export "${closure[@]}" > "$tmp/$system.nar"
+gzip "$tmp/$system.nar"
+artifactName=$system.nar.gz
+
+body='Automated release.
```

Comment:
Future: this is where the changelog goes.

```suggestion
body='# Changelog
* PR
* PR
* PR
etc
```